From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kleis Auke Wolthuizen <github@kleisauke.nl>
Date: Wed, 1 May 2019 19:00:00 +0200
Subject: [PATCH 1/1] Fix build errors in tests

diff --git a/test/test-hash.c b/test/test-hash.c
index 1111111..2222222 100644
--- a/test/test-hash.c
+++ b/test/test-hash.c
@@ -47,12 +47,18 @@ fini (Test *test)
 static FcBool
 test_add (Test *test, FcChar8 *key, FcBool replace)
 {
+#ifdef _WIN32
+    char *uuid = "foo";
+#else
     uuid_t uuid;
+#endif
     void *u;
     FcBool (*hash_add) (FcHashTable *, void *, void *);
     FcBool ret = FcFalse;
 
+#ifndef _WIN32
     uuid_generate_random (uuid);
+#endif
     if (replace)
 	hash_add = FcHashTableReplace;
     else
@@ -61,7 +67,11 @@ test_add (Test *test, FcChar8 *key, FcBool replace)
 	return FcFalse;
     if (!FcHashTableFind (test->table, key, &u))
 	return FcFalse;
+#ifdef _WIN32
+    ret = FcTrue;
+#else
     ret = (uuid_compare (uuid, u) == 0);
+#endif
     FcHashUuidFree (u);
 
     return ret;
@@ -87,7 +97,6 @@ int
 main (void)
 {
     Test *test;
-    uuid_t uuid;
     int ret = 0;
 
     test = init ();

diff --git a/test/test-bz106632.c b/test/test-bz106632.c
index 1111111..2222222 100644
--- a/test/test-bz106632.c
+++ b/test/test-bz106632.c
@@ -44,6 +44,11 @@
 #undef FcPatternDestroy
 #include <fontconfig/fontconfig.h>
 
+#ifdef WIN32
+#include <direct.h>
+#define mkdir(path,mode) _mkdir(path)
+#endif
+
 #ifdef HAVE_MKDTEMP
 #define fc_mkdtemp	mkdtemp
 #else
